lib/credo/check/warning/test_exs.ex

```diff
@@ -0,0 +1,67 @@
+defmodule Credo.Check.Warning.TestExs do
+  use Credo.Check,
+    base_priority: :high,
+    param_defaults: [excluded_paths: []],
+    explanations: [
+      check: """
+      Invoking mix test from the command line will run the tests in each file
+      matching the pattern `*_test.exs` found in the test directory of your project.
+
+      (from the `ex_unit` docs)
+
+      This check ensures that test files are not ending with `.ex` (which would cause them to be skipped).
+
+      """,
+      params: [
+        excluded_paths: "List of paths or regex to exclude from this check"
+      ]
+    ]
+
+  @test_files_with_ex_ending_regex ~r/test\/.*\/.*_test.ex$/
+
+  alias Credo.SourceFile
+
+  @doc false
+  def run(%SourceFile{filename: filename} = source_file, params \\ []) do
+    excluded_paths = Params.get(params, :excluded_paths, __MODULE__)
+    ignored = ignore_path?(filename, excluded_paths)
+
+    process_source_file(source_file, params, ignored)
+  end
+
+  defp process_source_file(_source_file, _params, true = _ignored), do: []
+
+  defp process_source_file(
+         %SourceFile{filename: filename} = source_file,
+         params,
+         false = _ignored
+       ) do
+    issue_meta = IssueMeta.for(source_file, params)
+
+    cond do
+      matches?(filename, @test_files_with_ex_ending_regex) ->
+        issue_for(issue_meta)
+        |> List.wrap()
+
+      :else ->
```

Comment:
We're using `true` as the catch all condition in the rest of the codebase.

```suggestion
      true ->
```